pkg/utils/kubernetes/metadata.go

```diff
@@ -30,6 +30,19 @@ func EnsureObjectMetaIsUpdated(
 ) (toUpdate bool, updatedMeta metav1.ObjectMeta) {
 	var metaToUpdate bool
 
+	// compare and enforce annotations. Take into account the fact that we don't
+	// want to compare all annotations as some might be added by other controllers.
+	// We only want to compare the annotations that are added by the operator.
+	for key, value := range generatedMeta.Annotations {
+		if existingValue, ok := existingMeta.Annotations[key]; !ok || existingValue != value {
+			if existingMeta.Annotations == nil {
+				existingMeta.Annotations = make(map[string]string, len(generatedMeta.Annotations))
+			}
```

Comment:
We should move this check upward as if Annotations is nil, the line above panics.

Reply:
Good catch, 02dc5eba1891d23e18b825dcd04a0643a0c0f071 PTAL